src/utils/dates.js

```diff
@@ -10,3 +12,101 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+export const handleNextPurchaseDate = (
+	name,
+	currentDate,
+	dateCreated,
+	purchasesCount,
+	dateNextPurchased,
+	dateLastPurchased,
+) => {
+	console.log('-------------------------------------------');
+	console.log(`[${name}]`);
+
+	try {
+		// get purchase intervals and get new estimation for next purchase date
+		const purchaseIntervals = calculatePurchaseIntervals(
+			currentDate,
+			dateCreated,
+			dateNextPurchased,
+			dateLastPurchased,
+		);
+		const estimatedNextPurchaseDate = getNextPurchaseEstimate(
+			purchaseIntervals,
+			purchasesCount,
+		);
+
+		console.log(
+			`New next purchase date: ${estimatedNextPurchaseDate.toLocaleString()}`,
+		);
+		console.log('-------------------------------------------');
+
+		return estimatedNextPurchaseDate;
+	} catch (error) {
+		throw new Error(`Failed getting next purchase date: ${error}`);
+	}
+};
+
+function getDaysBetweenDates(earlierDate, laterDate) {
```

Comment:
Would be helpful to add a comment a description above this helper, so intellisense can support hover over